### ISSUE TITLE
Feature Request: 

### ISSUE THREAD

**Issue by Muizzyranking** [chrisgrieser/nvim-rip-substitute#12]
### Checklist

- [X] I have read the README.
- [X] The feature would be useful to more users than just me.

### Feature Requested

I want to request for catpuccin intergration to make the rest of the code grey like when a section of code is highlighted just like when you use flash or leap. The selected part should retain the color while the rest of the code turn grey rather than the code code turning black. Check the image I uploaded. 
Sorry my description is not straightforward. 

I want to try this but I am thinking I need to fork catppuccin repo and add the integration but I decided to put this here because I think it would be cool if catpuccin has integration for this plugin, thanks.

### Relevant Screenshot

![image](https://github.com/chrisgrieser/nvim-rip-substitute/assets/106952550/638e62c7-9179-4d77-ac2b-42d88b4049c9)

**Comment by chrisgrieser** [comment]
Creating features for one specific theme is a bit too niche to add to a plugin, sorry.

**Comment by Muizzyranking** [comment]
Thank you for your response but do you know how I can achieve this? I know I would have to fork catppuccin and add the integration myself, but how do I know the hlgroup so I can do this myself?

**Comment by chrisgrieser** [comment]
The backdrop is actually a bit more complicated than just a highlight group. Basically, I achieve the effect by creating two dummy-windows that are fully black with transparency, that are laid upon the editor. Here is the function implementing it: https://github.com/chrisgrieser/nvim-rip-substitute/blob/014368a9cde67941170965235e97e73332b23d29/lua/rip-substitute/popup-win.lua#L144-L210

What you suggested is definitely possible, though that would require a very different logic applying highlights on all lines outside the range. So, I personally do not think that a slightly different aesthetic is worth writing all that code (though it's shorter than the current implementation), but if you end up writing that logic, I'd be open for a PR (so you do not have to maintain a fork for this)

**Comment by chrisgrieser** [comment]
Oh, and I just remembered: One non-aesthetic reason why I decided against creating a backdrop with the look you proposed is that it actually creates ambiguity in some edge cases. 

While `leap` and `flash` can simply paint the whole viewport in a comment color, using such a color for the area outside a range can be problematic when you have a fully commented line at the edge of the range. Since both, the outside-area and the commented line would then have the same appearance, it could become hard to distinguish between the two. Using the backdrop with the dummy-windows creates a "unique" appearance that does not leave such an ambiguity.
